src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -218,13 +220,20 @@ private Mono<SecurityAnalysisResult> run(SecurityAnalysisRunContext context, UUI
         return Mono.zip(network, contingencies)
                 .flatMap(tuple -> {
 
-                    Reporter reporter = context.getReportUuid() != null ? new ReporterModel("SecurityAnalysis", "Security analysis") : Reporter.NO_OP;
+                    Reporter rootReporter = Reporter.NO_OP;
+                    Reporter reporter = Reporter.NO_OP;
+                    if (context.getReportUuid() != null) {
+                        String rootReporterId = context.getReporterId() + "@" + AS_TYPE_REPORT;
+                        rootReporter = new ReporterModel(rootReporterId, rootReporterId);
+                        reporter = rootReporter.createSubReporter(AS_TYPE_REPORT, AS_TYPE_REPORT + " (${providerToUse})", "providerToUse", context.getProvider());
```

Comment:
Maybe check if context.getProvider() is null ?

Reply:
Done